native~/Runtime/src/CesiumMetadataValueImpl.cpp

```diff
@@ -305,824 +393,771 @@ getNativeMatValue(const DotNet::System::Object& object) {
+    F&& callback) {
   CesiumMetadataValueType valueType = value.valueType();
   assert(valueType.type == CesiumMetadataType::Mat4);
   switch (valueType.componentType) {
   case CesiumMetadataComponentType::Int8:
     return getNativeMatValue<glm::i8mat4x4, CesiumIntMat4x4>(
-        value.objectValue());
+        value.objectValue(),
+        std::forward<F>(callback));
   case CesiumMetadataComponentType::Uint8:
     return getNativeMatValue<glm::u8mat4x4, CesiumUintMat4x4>(
-        value.objectValue());
+        value.objectValue(),
+        std::forward<F>(callback));
   case CesiumMetadataComponentType::Int16:
     return getNativeMatValue<glm::i16mat4x4, CesiumIntMat4x4>(
-        value.objectValue());
+        value.objectValue(),
+        std::forward<F>(callback));
   case CesiumMetadataComponentType::Uint16:
     return getNativeMatValue<glm::u16mat4x4, CesiumUintMat4x4>(
-        value.objectValue());
+        value.objectValue(),
+        std::forward<F>(callback));
   case CesiumMetadataComponentType::Int32:
     return getNativeMatValue<glm::imat4x4, CesiumIntMat4x4>(
-        value.objectValue());
+        value.objectValue(),
+        std::forward<F>(callback));
   case CesiumMetadataComponentType::Uint32:
     return getNativeMatValue<glm::umat4x4, CesiumUintMat4x4>(
-        value.objectValue());
+        value.objectValue(),
+        std::forward<F>(callback));
   case CesiumMetadataComponentType::Int64:
     return getNativeMatValue<glm::i64mat4x4, CesiumIntMat4x4>(
-        value.objectValue());
+        value.objectValue(),
+        std::forward<F>(callback));
   case CesiumMetadataComponentType::Uint64:
     return getNativeMatValue<glm::u64mat4x4, CesiumUintMat4x4>(
-        value.objectValue());
+        value.objectValue(),
+        std::forward<F>(callback));
   case CesiumMetadataComponentType::Float32:
-    return getNativeMatValue<glm::mat4x4, float4x4>(value.objectValue());
+    return getNativeMatValue<glm::mat4x4, float4x4>(
+        value.objectValue(),
+        std::forward<F>(callback));
   case CesiumMetadataComponentType::Float64:
-    return getNativeMatValue<glm::dmat4x4, double4x4>(value.objectValue());
+    return getNativeMatValue<glm::dmat4x4, double4x4>(
+        value.objectValue(),
+        std::forward<F>(callback));
   default:
-    return std::monostate();
+    return callback(swl::monostate());
   }
 }
 
-CesiumMetadataValueImpl::ValueType
-getNativeStringValue(const DotNet::CesiumForUnity::CesiumMetadataValue& value) {
+template <typename F>
+auto getNativeStringValue(
+    const DotNet::CesiumForUnity::CesiumMetadataValue& value,
+    F&& callback) {
   DotNet::System::String string =
       CesiumForUnity::CesiumMetadataValue::GetObjectAsString(
           value.objectValue());
   if (string == nullptr) {
-    return std::monostate();
+    return callback(swl::monostate());
   }
 
-  return string.ToStlString();
+  return callback(string.ToStlString());
```

Comment:
Same here, is `std::move` omitted on purpose?

Reply:
That's already an rvalue (because ToStlString returns by value), no need for move there.